### ISSUE TITLE
datatable - paginator is not recalculated after filtering

### ISSUE THREAD

**Issue by UmbarPantera** [primefaces/primereact#181]
Filtering might decreases the number of rows of your table. There should be a check, if the number of needed pages has decreased. If the number has decreased, the paginator needs to be recalculated.
At the moment it seems the paginator is not recalculated. Because of this the paginator might show some pages/page numbers, that no longer exist/ are empty.

**Comment by ebonventi** [comment]
The problem happened with backend or frontend filter?

**Comment by ebonventi** [comment]
@UmbarPantera the number of page is calculate  from  totalRecord attribute, when you are using a frontend filter (lazyload false) the datable do this for you, but if lazyload is on, then you need set manualy the new value of totalRecord attribute.

**Comment by UmbarPantera** [comment]
@ebonventi 
Lazy is default/false and in theory everything should work, but for unknow reasons it doesn't.  The table is filtered, but totalPages is not adjusted.

**Comment by ebonventi** [comment]
@UmbarPantera can you share your source code to i try to analyze and help?

**Comment by UmbarPantera** [comment]
@ebonventi 
I was just testing the datatable. Only 2 columns. 50 rows of data.

`<DataTable value={this.state.dataArray} paginator={true} rows={10}>`
   `<Column field={'numbr'} header={'Number'} filter={true}/>`
   `<Column field={'name'} header={'Name'} filter={true}/>`
 `</DataTable>`


![pr-pagfail](https://user-images.githubusercontent.com/31218856/32953101-75a01c78-cbaf-11e7-882b-0b3f593c86f0.jpg)

**Comment by cagataycivici** [comment]
We could not see the problem, can you replicate at;

https://www.primefaces.org/primereact/#/datatable/filter

**Comment by UmbarPantera** [comment]
Your filter example works.

I mad some tests with my data in your example. Works too. I used v1.1.0, like I did in, when I got the bug. Worked.

Then I started from scratch again. 

I used https://github.com/primefaces/primereact-typescript-quickstart to start and did the same changes like I did last time. The bug happens again. Bug happens again. DataTable is not filtered.

Hier is the whole modified app-file:
[App.zip](https://github.com/primefaces/primereact/files/1511061/App.zip)
